### DISCUSSION TITLE
ShortHand Template Feature Request

### DISCUSSION THREAD

**Discussion by aashish803** [OrchardCMS/OrchardCore#16180] [Q&A]
Hey Everyone, 
Good Morning 

I need to know that does Orchard provide a feature or a module to Integrate Short Hand Templates [ https://app.shorthand.com/ ]. 
I need the feature of Short Hand Template in orchard core. Can you guys tell me how I implement that?

Thanks
Regards

**Comment by Piedone** [comment]
It doesn't provide that out of the box, nor as a third-party feature as far as I'm aware. What kind of integration are you looking for, exactly, what's your use case?

**Reply by aashish803** [reply]
Okay @Piedone 
Let me explain you a bit more about my requirement. If you have looked / worked on wordpress elementor , how they provide templates for every section [ eg :- banner section, they provide different layout for banner section and user can just edit the settings of those templates like can change background image, can change heading text , heading font, etc ] User does not need to write code for that. 
I need to replicate the functionality of that in our orchard core , I want to know does Orchard already provide any such functionality or feature. Or do we need to make from the scratch?
Thanks 
Regards

**Reply by Piedone** [reply]
OC doesn't have a direct equivalent but it does provide such features in a different way:

- Foremost, you'd build custom layouts with [Flow Part](https://docs.orchardcore.net/en/latest/reference/modules/Flow/), and perhaps add custom widget types for your users to simplify common scenarios.
- The [`OrchardCoreContrib.Html` module](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/blob/main/src/OrchardCoreContrib.Html/README.md) lets you use the [GrapeJS editor](https://grapesjs.com/) to generate HTML.
- If Shorthand or any other editor can be embedded in an ASP.NET Core web app, and in the end, generates HTML (and CSS/JS) then it can be integrated into OC as simply as `OrchardCoreContrib.Html` does it for GrapeJS (check out the two files in its Views folder, that's it).

**Reply by aashish803** [reply]
Thanks for responding.
I have tried using GrapeJS editor](https://grapesjs.com/) editor but  I think that is very much similar to figma & adobe illustrator. If I am not wrong. 
But what I am asking is completely different.

**Reply by Piedone** [reply]
OK then!
